src/async_fuse/memfs/cache/memory_cache/cache.rs

```diff
@@ -65,13 +67,46 @@ impl<P, S> MemoryCache<P, S> {
             policy,
             backend,
             block_size,
-            write_through,
+            write_through: AtomicBool::new(write_through),
             pending_write_back: RwLock::default(),
             command_sender,
             truncate_records: HashMap::new(),
         }
     }
 
+    /// Check if the cache is running on wtire through policy.
+    fn write_through(&self) -> bool {
+        self.write_through.load(Ordering::Acquire)
+    }
+
+    /// Fallback to write through.
+    async fn fallback(&self)
+    where
+        S: Storage,
+    {
+        self.write_through.store(true, Ordering::Release);
+
+        // Ensure that all pending blocks are flushed to the backend.
+        let pending = mem::take(&mut *self.pending_write_back.write().await);
```

Comment:
`mem::take` actually copy the whole data struture out to the stack. Is that what you really want?

Reply:
I guess what you want is "release the lock as soon as possible".

Reply:
> I guess what you want is "release the lock as soon as possible".

Yes. And `mem::take` should do shallow copy (in move semantics), so I think that to `take` a `HashMap` is OK.